cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -409,13 +455,45 @@ private JSONObject normalizeRemotePrincipal(final OpenScienceFrameworkCredential
         final DOMSource source = new DOMSource(document);
         final StringWriter writer = new StringWriter();
         final StreamResult result = new StreamResult(writer);
-
         this.institutionsAuthTransformer.transform(source, result);
 
         // convert transformed xml to json
         return XML.toJSONObject(writer.getBuffer().toString());
     }
 
+    /**
+     * Update `credential` with remote principal authenticated by external CAS.
+     *
+     * @param credential the credential whose authenticationHeaders is going to be udpated
+     * @param principal the principal created by external CAS authentication
+     * @return the updated credential
+     * @throws AccountException RemoteUserFailedLoginException
+     */
+    private OpenScienceFrameworkCredential constructCredentialFromRemotePrincipal(
```

Comment:
Update `credential.authenticationHeaders` which will be normalized with XML XSLT.
- `Shib-Identity-Provider`
- `uid`
- attributes in `principal`
  - OK State University
    - `mail` is mapped to `username` (they don't provide `eppn`)
    - `fullname` is mapped to `uid` (they don't provider `displayName`)
    - `sn` is mapped to `familyName`
    - `givenName` is mapped to `givenName`